Source/inv.cpp

```diff
@@ -325,14 +325,158 @@ int FindTargetSlotUnderItemCursor(Point cursorPosition, Size itemSize)
 	return NUM_XY_SLOTS;
 }
 
-void CheckInvPaste(Player &player, Point cursorPosition)
+void ChangeBodyEquipment(int slot, Player &player, item_equip_type il)
 {
-	Size itemSize = GetInventorySize(player.HoldItem);
+	auto iLocToInvLoc = [&slot](item_equip_type loc) {
+		// Table?
+		switch (loc) {
+		case ILOC_HELM:
+			return INVLOC_HEAD;
+		case ILOC_RING:
+			return (slot == SLOTXY_RING_LEFT ? INVLOC_RING_LEFT : INVLOC_RING_RIGHT);
+		case ILOC_AMULET:
+			return INVLOC_AMULET;
+		case ILOC_ARMOR:
+			return INVLOC_CHEST;
+		default:
+			app_fatal("Unexpected equipment type");
+		}
+	};
+	const auto body_loc = iLocToInvLoc(il);

```

Comment:
Since you do not need the lambda it self but just the result I would suggest executing it right after creating it:

```suggestion
	const auto body_loc = [&slot](item_equip_type loc) {
		// Table?
		switch (loc) {
		case ILOC_HELM:
			return INVLOC_HEAD;
		case ILOC_RING:
			return (slot == SLOTXY_RING_LEFT ? INVLOC_RING_LEFT : INVLOC_RING_RIGHT);
		case ILOC_AMULET:
			return INVLOC_AMULET;
		case ILOC_ARMOR:
			return INVLOC_CHEST;
		default:
			app_fatal("Unexpected equipment type");
		}
	}(il);
```